### PR TITLE
Shell grain: check %COMSPEC% for the shell for Windows

### PR DESCRIPTION
**Pull request by morganwillcock**
### What does this PR do?
Return the command processor instead of /bin/sh for the shell grain on Windows

### What issues does this PR fix or reference?
#43506 

### Previous Behavior
Windows returns '/bin/sh' as its shell

### New Behavior
Windows expands the environment variable `%COMSPEC%` to identify the shell, or returns 'C:\Windows\system32\cmd.exe' as the default.

### CODE REVIEW

**Comment by garethgreenaway** [comment]
Nice!

File: salt/grains/extra.py [hunk 1]
```diff
@@ -21,7 +22,14 @@ def shell():
     '''
     # Provides:
     #   shell
-    return {'shell': os.environ.get('SHELL', '/bin/sh')}
+    if salt.utils.platform.is_windows():
+        env_var = 'COMSPEC'
+        default = r'C:\Windows\system32\cmd.exe'
```
Comment (damon-atkins) [inline salt/grains/extra.py]: Maybe something like this os.getenv('ComSpec', '{0}\\system32\\cmd.exe'.format(os.getenv('WINDIR')))

**Comment by damon-atkins** [comment]
Thanks

File: salt/grains/extra.py (hunk 1 above)
Comment (morganwillcock) [reply salt/grains/extra.py]: I did consider it, but thought if COMSPEC isn't set (system would be very broken) there is no guarantee that WINDIR is set either. So this way at least the path is valid, i.e. /bin/sh might not exist either.

**Review by cachedout — APPROVED** [review]

**Pull request merged**